sdk/daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SBuiltinInterfaceTest.scala

```diff
@@ -24,28 +23,160 @@ import org.scalatest.Inside
 import org.scalatest.freespec.AnyFreeSpec
 import org.scalatest.matchers.should.Matchers
 import org.scalatest.prop.TableDrivenPropertyChecks
+import com.digitalasset.daml.lf.interpretation.{Error => IE}
 
-import util.{Failure, Success, Try}
+import scala.util.{Failure, Success, Try}
 
-class SBuiltinInterfaceTestV2 extends SBuiltinInterfaceTest(LanguageMajorVersion.V2)
+class SBuiltinInterfaceTestDefaultLf
+    extends SBuiltinInterfaceTest(
+      LanguageVersion.default,
+      Compiler.Config.Default(LanguageMajorVersion.V2),
+    )
+class SBuiltinInterfaceTestDevLf
+    extends SBuiltinInterfaceTest(
+      LanguageVersion.v2_dev,
+      Compiler.Config.Dev(LanguageMajorVersion.V2),
+    )
+
+class SBuiltinInterfaceUpgradeTest extends AnyFreeSpec with Matchers with Inside {
+
+  import EvalHelpers._
+
+  val alice = Ref.Party.assertFromString("Alice")
 
-class SBuiltinInterfaceTest(majorLanguageVersion: LanguageMajorVersion)
+  // The following code defines a package -iface-pkg- that defines a single interface Iface.
+  val ifacePkgName = Ref.PackageName.assertFromString("-iface-pkg-")
+  val ifacePkgId = Ref.PackageId.assertFromString("-iface-package-id-")
```

Comment:
extremely picky comment. 
why `-iface-pkg-` and `-iface-package-`

Reply:
Done